File: atividades/c/compara_pares.c
#include <stdio.h>

int main(){
    char entrada[20];
    int x, y;

    int menor_soma = 0, maior_soma = 0;
    int soma_dos_pares = 0;
    int soma_total = 0;
    int quantidade_pares = 0;
    int quantidade_valores = 0;

    printf("Digite as entradas em pares (ex: 12 3, 3 4, 9 3). Digite * para parar. \n");

    while (1) {
        scanf("%s", entrada);

        if(entrada[0] == '*') {
            break;
        }

        sscanf(entrada, "%d,%d", &x, &y);

        int soma = x + y;

        if(quantidade_pares == 0){
            menor_soma = soma;
            maior_soma = soma;
        } else {
            if(soma < menor_soma) menor_soma = soma;
            if(soma > maior_soma) maior_soma = soma;
        }

        soma_dos_pares += soma;
        soma_total += x + y;
        quantidade_pares++;
        quantidade_valores += 2;
    }

    float media_pares = (float)soma_dos_pares / quantidade_pares;
    float media_total = (float)soma_total / quantidade_valores;

    printf("menor soma: %d\n", menor_soma);
    printf("maior soma: %d\n", maior_soma);
    printf("media dos pares: %.2f\n", media_pares);
    printf("media de todos: %.2f\n", media_total);

    return 0;
}

File: atividades/java/ComparaPares.java
import java.util.Scanner;

public class ComparaPares {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int menorSoma = 0, maiorSoma = 0;
        int somaDosPares = 0;
        int somaTotal = 0;
        int quantidadePares = 0;
        int quantidadeValores = 0;

        System.out.println("Digite as entradas em pares (ex: 12,3). Digite * para parar.");

        while(true){
            String entrada = scanner.nextLine();

            if(entrada.equals("*")){
                break;
            }

            String[] partes = entrada.split(",");
            if(partes.length != 2){
                System.out.println("Entrada inválida. Use o formato x,y.");
                continue;
            }

            int x = Integer.parseInt(partes[0].trim());
            int y = Integer.parseInt(partes[1].trim());

            int soma = x + y;

            if (quantidadePares == 0){
                menorSoma = soma;
                maiorSoma = soma;
            } else {
                if (soma < menorSoma) menorSoma = soma;
                if (soma > maiorSoma) maiorSoma = soma;
            }

            somaDosPares += soma;
            somaTotal += x + y;
            quantidadePares++;
            quantidadeValores += 2;
        }

        if(quantidadePares > 0){
            double mediaPares = (double) somaDosPares / quantidadePares;
            double mediaTotal = (double) somaTotal / quantidadeValores;

            System.out.println("Menor soma: " + menorSoma);
            System.out.println("Maior soma: " + maiorSoma);
            System.out.printf("Média dos pares: %.2f\n", mediaPares);
            System.out.printf("Média de todos os valores: %.2f\n", mediaTotal);
        } else {
            System.out.println("Nenhum par válido foi inserido.");
        }

        scanner.close();
    }
}